app/src/main/java/com/jerboa/ui/components/common/TextBadge.kt

```diff
@@ -33,12 +39,59 @@ fun TextBadge(
         Text(
             text = text,
             style = textStyle,
-            overflow = TextOverflow.Clip,
             maxLines = 1,
             color = textColor,
             modifier =
                 Modifier
-                    .padding(horizontalTextPadding.dp, verticalTextPadding.dp),
+                    .padding(horizontalTextPadding.dp, verticalTextPadding.dp)
+                    .basicMarquee(),
         )
     }
 }
+
+/**
+ * Displays activitypub items (communities, users), with a smaller @instance shown
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ItemAndInstanceTitle(
+    modifier: Modifier = Modifier,
+    title: String,
+    actorId: String?,
+    local: Boolean,
+    itemColor: Color = MaterialTheme.colorScheme.primary,
+    itemStyle: TextStyle = MaterialTheme.typography.labelLarge,
+    instanceColor: Color = MaterialTheme.colorScheme.onSurface.muted,
+    instanceStyle: TextStyle = MaterialTheme.typography.labelSmall,
+) {
+    val serverStr = if (!local) {
+        actorId?.let { aId ->
+            "@${hostName(aId)}"
+        } ?: run {
+            null
+        }
+    } else {
+        null
+    }
+
+    Text(
+        text = buildAnnotatedString {
```

Comment:
Not sure how expensive this is but i would encase this in a remember with the color and title as key

Reply:
I could do that but I don't see why its necessary, since its state never changes within this composable.

Reply:
Well it shouldn't if the composable is skippable. But I haven't checked yet that it is. Else it would recompose each time anyway even if the state didn't change